src/components/templates/ScreenShare/agoraHooks/useAgoraCamera.tsx

```diff
@@ -0,0 +1,88 @@
+import { useState, useEffect, useCallback } from "react";
+import AgoraRTC, {
+  IAgoraRTCClient,
+  ILocalAudioTrack,
+  ILocalVideoTrack,
+} from "agora-rtc-sdk-ng";
+
+export default function useAgoraCamera(
+  client: IAgoraRTCClient | undefined,
+  channel: { appId: string; channel: string; token: string }
+): {
+  localCameraTrack: ILocalVideoTrack | undefined;
+  toggleCamera: () => void;
+  toggleMicrophone: () => void;
+  isCameraOn: boolean;
+  closeTracks: () => void;
+  isMicrophoneOn: boolean;
+  onJoin: () => void;
+} {
+  const [localCameraTrack, setLocalCameraTrack] = useState<
+    ILocalVideoTrack | undefined
```

Comment:
I don't think you need to explicitly specify it's undefined here

Reply:
Yup. `useState<ILocalVideoTrack>()` is sufficient here.